src/microbe_stage/systems/ProcessSystem.cs

```diff
@@ -427,73 +427,76 @@ public static (float Production, float Consumption) CalculateOrganelleATPBalance
             }
             else
             {
-                processData = CalculateProcessMaximumSpeed(process, biome, amountType);
+                processData = CalculateProcessMaximumSpeed(process, biome, amountType, true);
             }
 
-            if (processData.WritableInputs.TryGetValue(ATP, out var amount))
+            if (processData.WritableInputs.TryGetValue(ATP, out var amount) && amount > 0)
             {
                 processATPConsumption += amount;
 
                 result?.AddConsumption(organelle.Definition.InternalName, amount);
             }
 
-            if (processData.WritableOutputs.TryGetValue(ATP, out amount))
+            if (processData.WritableOutputs.TryGetValue(ATP, out amount) && amount > 0)
             {
                 result?.AddProduction(organelle.Definition.InternalName, amount);
 
-                var isInPatch = true;
+                processATPProduction += amount;
+            }
+        }
+
+        return (processATPProduction, processATPConsumption);
+    }
 
-                foreach (var input in processData.WritableInputs)
+    /// <summary>
+    ///   Calculates the maximum speed a process can run at in a biome based on the environmental compounds.
+    ///   Can be switched between the average, maximum etc. conditions that occur in the span of an in-game day.
+    /// </summary>
+    public static ProcessSpeedInformation CalculateProcessMaximumSpeed(TweakedProcess process,
+        BiomeConditions biome, CompoundAmountType pointInTimeType, bool checkIfInPatch)
```

Comment:
I think some other name for the parameter is needed. Something like `bool requireInputCompoundsInBiome`?